recipes/CommonVoice/common_voice_prepare.py

```diff
@@ -325,34 +352,121 @@ def galc(w):
     logger.info(msg)
 
 
-def check_commonvoice_folders(data_folder):
-    """
-    Check if the data folder actually contains the Common Voice dataset.
 
-    If not, raises an error.
 
+def check_commonvoice_folders(data_folder,dataset_version,language):
+    """
+    Check if the data folder actually contains the Common Voice dataset.
+    If not, download the data 
     Returns
     -------
     None
-
-    Raises
-    ------
-    FileNotFoundError
-        If data folder doesn't contain Common Voice dataset.
     """
 
     files_str = "/clips"
 
+
     # Checking clips
+   
     if not os.path.exists(data_folder + files_str):
-
-        err_msg = (
+        logger.info(
             "the folder %s does not exist (it is expected in "
             "the Common Voice dataset)" % (data_folder + files_str)
         )
-        raise FileNotFoundError(err_msg)
+        logger.info(f"Downloading dataset for {language}")
+        download_locale(language,data_folder,dataset_version)
 
 
+    else:
+        logger.info("Dataset already downloaded")
+
+def download_locale(
```

Comment:
What's your thoughts on that @TParcollet @anautsch? Should we have this function in SpeechBrain?

Reply:
I have discussed a little bit with @TParcollet, it is a neat feature to have, but the user should be aware of this. He should explicitly ask to download the dataset. So, I think adding a variable in the YaMLs files, e.g. `download_common_voice: True` would be enough. If this value is set to `True` then you automatically download common_voice if the folder is not found, else, you warn the user that the folder hasn't been find.

Reply:
Having several thoughts here. For LibriSpeech recipes, there's an auto-download functionality (not bad to have one here). Yet, LibriSpeech doesn't change as frequent in their versioning (that download script produces the same outcome all the time; might not be the case here). The default option should be not to download, but to check & to hint that downloading would be possible.

Does this download function produce the same file structure as we are used to have w/o it? 
Are there differences? (if not, more consistency is good.)

The concerns I'd see are towards reproducibility & keeping internet traffic at the lowest demand necessary (not everyone needs the latest CommonVoice version, always). Managing their versions might be even an overhead to what SB recipes need to handle. Hope we can keep overheads out of SB if these are special cases to SB-externalities.

What's on your mind?

Reply:
It doesn't downloads the latest version. We pass which version to download and it downloads that specific version of common-voice which  produces the same outcome all the time. We always check If they already downloaded the data , if so we skip the downloading.

Reply:
i removed the downloaing part